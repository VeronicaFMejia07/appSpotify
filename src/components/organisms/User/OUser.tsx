import MUser from 'components/molecules/User/MUser';
import React, { FC, useEffect, useState } from 'react'
import { useAppDispatch } from 'redux/hooks/hooks';
import { setUser } from 'redux/slices/userSlice';
import { getUser } from 'redux/thunks/userThunk';
import { User } from 'utils/interfaces/User/IUser';

interface IProps {
  userInf: User;
}

const OUser: FC= ({ }) => {
  const [user, setUserView]=useState<User>();
  const dispatch=useAppDispatch();

  useEffect(()=>{
    getUser().then(data=>{
      dispatch(setUser(data));//Se trae de redux el reducer
      setUserView(data);//Se asigna a la variable la data que trae la petici√≥n
    })
  },[dispatch])
  
  return (
    <div>
      {
        <MUser nameUser={user?.display_name} imgUser={user?.images[0].url} />
      }
    </div>
  )
}

export default OUser