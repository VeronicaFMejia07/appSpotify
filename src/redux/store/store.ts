import { configureStore } from "@reduxjs/toolkit";
import {userSlice} from "redux/slices/userSlice";
import {playlistSlice } from "redux/slices/playlistSlice";
import {favoritesSlice} from "redux/slices/favoritesSlice";

export const store= configureStore({
    reducer:{
        playlist: playlistSlice.reducer,
        favorites: favoritesSlice.reducer,
        user: userSlice.reducer,
    }
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export default store