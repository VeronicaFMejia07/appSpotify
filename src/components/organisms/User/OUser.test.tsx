import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from 'redux/store/store';
import OUser from './OUser';

//Describir la prueba que se este haciendo
describe('Pruebas en OUser', () => {
    test('Renderiza componente--OUser', () => {
        const { container } = render(
            <Provider store={store}>
                <OUser />
            </Provider>
        )

        //Evalua que el container sea igual a lo que contiene el componente
        expect(container).toMatchSnapshot();
    })

})


