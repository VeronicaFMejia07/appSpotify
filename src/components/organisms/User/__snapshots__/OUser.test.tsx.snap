// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en OFavoritesList Renderiza componente--OFavoritesList 1`] = `
<div>
  <div>
    <header
      class="header"
    >
      <div
        role="contenedor"
      >
        <p
          class="header__name"
        />
      </div>
      <figure
        class="header__figure"
      >
        <figure
          class="main__figure"
          role="contenedor"
        >
          <img
            alt="spotify"
            class="header__img"
          />
        </figure>
      </figure>
    </header>
  </div>
</div>
`;

exports[`Pruebas en OUser Renderiza componente--OFavoritesList 1`] = `
<div>
  <div>
    <header
      class="header"
    >
      <div
        role="contenedor"
      >
        <p
          class="header__name"
        />
      </div>
      <figure
        class="header__figure"
      >
        <figure
          class="main__figure"
          role="contenedor"
        >
          <img
            alt="spotify"
            class="header__img"
          />
        </figure>
      </figure>
    </header>
  </div>
</div>
`;

exports[`Pruebas en OUser Renderiza componente--OUser 1`] = `
<div>
  <div>
    <header
      class="header"
    >
      <div
        role="contenedor"
      >
        <p
          class="header__name"
        />
      </div>
      <figure
        class="header__figure"
      >
        <figure
          class="main__figure"
          role="contenedor"
        >
          <img
            alt="spotify"
            class="header__img"
          />
        </figure>
      </figure>
    </header>
  </div>
</div>
`;
